### PR TITLE
add the log-collector to the utils folder and update README

### PR DESCRIPTION
**Pull request by jeremybumsted**
@moskyb This will add a handy little script that I demoed recently, which should help customers collect the logs we need in Support. If you want to chat about it or have suggestions or anything let me know 😄

### CODE REVIEW

**Review by moskyb — APPROVED** [review]
LGTM! Great work

File: utils/log-collector [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+pwd=$(pwd)
+dir=$(mktemp -d "$pwd"/buildkite-logs-"$now")
+
+echo "Collecting logs for instance $instanceId"
+
+# Collect the logs for the instance
+aws logs get-log-events --log-group-name /buildkite/buildkite-agent --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/buildkite-agent-"$instanceId".log
+aws logs get-log-events --log-group-name /buildkite/system --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/system-"$instanceId".log
+aws logs get-log-events --log-group-name /buildkite/lifecycled --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/lifecycled-"$instanceId".log
+aws logs get-log-events --log-group-name /buildkite/docker-daemon --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/docker-daemon-"$instanceId".log
```
Comment (pda) [inline utils/log-collector:66]: I like the simplicity/directness, and would avoid any unnecessary complexity that would make this script harder to read/understand, but might make it _easier_ to read/understand?

```bash
for name in buildkite-agent system lifecycled docker-daemon; do
  aws logs get-log-events \
    --log-group-name "/buildkite/$name" \
    --log-stream-name "$instanceId" \
    --output "$outputFormat" \
    >> "$dir/$name-$instanceId.log"
end
```

File: utils/log-collector [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+lambdaLogGroup=$(aws cloudformation describe-stack-resources --stack-name "$lambdaFunction" --logical-resource-id LogGroup --query "StackResources[*].PhysicalResourceId" --output text)
+logStreams=$(aws logs describe-log-streams --log-group-name "$lambdaLogGroup" --order-by LastEventTime --query "logStreams[?creationTime > \`$logAge\`] | [*].logStreamName" --output text | tr -s '[:space:]' '\n')
+
+# Iterate through all of the log streams we collected from the lambda log group
+
+echo "Collecting Lambda logs for $lambdaFunction"
+for i in $logStreams
+do
+    fileName=$(echo "$i" | sed -r 's/\//-/g')
+    aws logs get-log-events --log-group-name "$lambdaLogGroup" --log-stream-name "$i" --output "$outputFormat" >> "$dir"/lambda-"$fileName".log
```
Comment (pda) [inline utils/log-collector:85]: I personally find this style easier to read:

```diff
-"$dir"/lambda-"$fileName".log
+"$dir/lambda-$fileName.log"
```

Or if you're worried about differentiating the vars from other content:

```diff
-"$dir"/lambda-"$fileName".log
+"${dir}/lambda-${fileName}.log"
```

File: utils/log-collector [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+done
+
+# Collect ASG activity for the scaling group
+autoScalingGroup=$(aws cloudformation describe-stack-resources --stack-name "$stackName" --logical-resource-id AgentAutoScaleGroup --query "StackResources[*].PhysicalResourceId" --output text)
+echo "Collecting ASG activity for $autoScalingGroup"
+aws autoscaling describe-scaling-activities --auto-scaling-group-name "$autoScalingGroup" >> "$dir"/asg-activities-"$stackName".log
+
+# Make an archive with the logs
+zip -jq buildkite-logs-"$now".zip "$dir"/*.log
```
Comment (pda) [inline utils/log-collector:94]: I wonder if you'll have trouble with `zip` availability.

macOS has it, but e.g. The minimal linux docker containers don't tend to have it:

```
pda@paulbookpro ~/bk/buildkite $ for img in debian ubuntu archlinux centos alpine; do echo -n "$img: "; docker run --rm $img sh -c "zip"; done
debian: sh: 1: zip: not found
ubuntu: sh: 1: zip: not found
archlinux: sh: line 1: zip: command not found
centos: sh: zip: command not found
alpine: sh: zip: not found
```

Tar might be a safer assumption…


```
pda@paulbookpro ~/bk/buildkite $ for img in debian ubuntu archlinux centos alpine; do echo -n "$img: "; docker run --rm $img sh -c "tar --version | head -n1"; done
debian: tar (GNU tar) 1.34
ubuntu: tar (GNU tar) 1.30
archlinux: tar (GNU tar) 1.34
centos: tar (GNU tar) 1.30
alpine: tar (busybox) 1.34.1
```

But maybe people are mostly doing this from fully-optioned macOS / Linux machines that are likely to have `zip`.

**Review by pda — COMMENTED** [review]
Excellent 👍🏼
Some non-urgent comments.

**Pull request merged**